sensor_main.py

```
@@ -0,0 +1,141 @@
+        for timer in self.sensor_timers.values():
+            timer.start()
+
+        self.config_timer.start()
+
+    def _update_freq(self):
+        stats = Path(self.config_file).stat()
+        if stats.st_mtime != self.last_mtime:
+            self.last_mtime = stats.st_mtime
+            for sensor_type, new_freq in self._get_current_conf().items():
+                self.sensor_timers[sensor_type].interval = new_freq
+
+    def stop(self):
+        self.stop_event.set()
+
+    def run(self):
+        self._initialize()
+        self.stop_event.wait()
+        self.close()
+
+    def close(self):
+        for switch in self.switches:
+            switch.clean()
+        for timer in self.sensor_timers.values():
+            timer.cancel()
+            timer.join()
+        self.config_timer.cancel()
+        self.config_timer.join()
+
+        self.database.close()
+
+    def _get_current_conf(self) -> dict[SensorType, float]:
+        config = configparser.ConfigParser()
+        config.read(self.config_file)
+        sensor_conf = {}
+        for sensor_type in SensorType:
+            type_conf = SensorType.to_conf(sensor_type)
+            if type_conf in config['sensors_config']:
+                sensor_conf[sensor_type] = float(config['sensors_config'][type_conf])
+        return sensor_conf
+
+    def _get_sensor_timers(self, readings: SensorReadings, interface: Interface):
+        dht = DHT()
+        bmp = BMP280()
+        pmsa = PMSA003C()
+        start_conf = self._get_current_conf()
+
+        def update_reading(sensor: Sensor, sensor_type: SensorType):
+            try:
+                value = sensor.get_reading(sensor_type)
+                readings.add(sensor_type, value)
+                interface.update_sensor(sensor_type)
+            except SensorReadingError:
+                logging.warning("SensorReadingError: %s, %s", sensor.__class__.__name__, sensor_type.name)
+
+        #pylint: disable=invalid-name
```

Align rather then suppress errors?